### PR TITLE
Protobuf version change to 3.9

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Protobuf version change from 3.7.0 to 3.9.0.  Fixes issue #721 .

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.

### CODE REVIEW

File: cmake/Protobuf.cmake [hunk 1]
```diff
@@ -1,132 +1,239 @@
...
+  ${protobuf_source_dir}/src/google/protobuf/compiler/objectivec/objectivec_extension.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/objectivec/objectivec_field.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/objectivec/objectivec_file.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/objectivec/objectivec_generator.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/objectivec/objectivec_helpers.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/objectivec/objectivec_map_field.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/objectivec/objectivec_message.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/objectivec/objectivec_message_field.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/objectivec/objectivec_oneof.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/objectivec/objectivec_primitive_field.cc
```
Comment (hewigovens) [inline cmake/Protobuf.cmake]: do we need objective c? previously we don't include `compiler` folder?

File: cmake/Protobuf.cmake [hunk 2]
```diff
@@ -1,132 +1,239 @@
...
+  ${protobuf_source_dir}/src/google/protobuf/compiler/cpp/cpp_generator.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/cpp/cpp_helpers.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/cpp/cpp_map_field.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/cpp/cpp_message.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/cpp/cpp_message_field.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/cpp/cpp_padding_optimizer.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/cpp/cpp_primitive_field.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/cpp/cpp_service.cc
+  ${protobuf_source_dir}/src/google/protobuf/compiler/cpp/cpp_string_field.cc
+  #${protobuf_source_dir}/src/google/protobuf/compiler/csharp/csharp_doc_comment.cc
```
Comment (hewigovens) [inline cmake/Protobuf.cmake]: just remove it

**Comment by hewigovens** [comment]
when you squash and merge, make sure type a short git summary (31 commits) :)

**Comment by catenocrypt** [comment]
Ready for review.  Status:
- builds (macos, ios, android) OK locally
- builds (linux, ios, android) OK in CI builds
- iOS podfile tested with dummy client project

**Comment by catenocrypt** [comment]
> looks cleaner, we only build protobuf once, correct?

No, I backtracked on that.  The CMake build is used not only for MacOS/Linux, but for iOS/Android builds as well, and each one has its own special settings.  So one build in install-dependencies is not enough.  But that one is also needed.  The only things I could do is some cleanup, e.g. tests/CMakeFile was looking for pre-built protobuf lib, when in fact it was linking with the one built in the same CMake flow.

**Pull request merged**